src/app/util/basic-types.h

```diff
@@ -31,13 +31,16 @@
 typedef uint8_t CHIPEndpointId;
 typedef uint16_t CHIPClusterId;
 typedef uint16_t CHIPAttributeId;
+typedef uint16_t CHIPGroupId;
```

Comment:
I guess this is fine.

It just points out the fact that, in the current shape, we will need to do string based matching on the name of the arguments in the generator since there is no notion of `GroupId`. By notion I mean something like `attribId` in the section _2.6.2.2 Data Type Table_ of the ZCL specification.

Concretely the code will need to match **groupId** and some variants such as **groupIdFrom** and **groupIdTo**

Or (better) we will need to fix the Silabs XML to use a special type, even if it is not a special type such as `attribId` in order to not have to do string based matching. By special type I'm thinking of something like [SGroupId](https://github.com/project-chip/zap/blob/0b5940ff1af4f8e6ed0e54da79f25156a1210e11/zcl-builtin/dotdot/Scenes.xml#L32) which is used to mark the _Scene_ plugin arguments that uses a group identifier but with some constraints on their values.

Reply:
Hmm.  Yes, we might need to do some work on the generator side to make the switching simple....